compiler/codegen/OMRCodeGenPhase.cpp

```diff
@@ -432,13 +432,64 @@ OMR::CodeGenPhase::performSetupForInstructionSelectionPhase(TR::CodeGenerator *
       std::map<TR::TreeTop*, TR::TreeTop*, std::less<TR::TreeTop*>, TR::typed_allocator<std::pair<TR::TreeTop*, TR::TreeTop*>, TR::Allocator> >
          currentTreeTopToappendTreeTop(std::less<TR::TreeTop*> (), mapAllocator);
 
+      TR_BitVector *unAnchorableAloadiNodes = comp->getBitVectorPool().get();
+
       for (TR::PreorderNodeIterator iter(comp->getStartTree(), comp); iter != NULL; ++iter)
          {
          TR::Node *node = iter.currentNode();
 
          traceMsg(comp, "GuardedStorage: Examining node = %p\n", node);
 
-         if (node->getOpCodeValue() == TR::aloadi)
+         if (node->getOpCodeValue() == TR::NULLCHK)
```

Comment:
I made the change in 8f58ce352176a6e0aa1ce6a1db876373e320191b to check for the other *AndNULLCHK scenarios.